src/solvers/smt2_incremental/smt_solver_process.cpp

```diff
@@ -28,6 +30,15 @@ void smt_piped_solver_processt::send(const smt_commandt &smt_command)
   const std::string command_string = smt_to_smt2_string(smt_command);
   log.debug() << "Sending command to SMT2 solver - " << command_string
               << messaget::eom;
+
+  if(out_stream != nullptr)
+  {
+    // Using std::endl to flush the stream as it is a debugging functionality,
+    // and we can guarantee a consistent output in case of hanging after
+    // (check-sat)
+    *out_stream << command_string << std::endl;
```

Comment:
Do you perhaps just want to use `std::flush` instead? Most of the comment would remain useful to keep, though.

Reply:
As we want a new line and a `flush` I prefer `std::endl` instead of adding a new line + adding an explicit `std::flush`.

Reply:
Ok, but then you might want the comment to say "Using std::endl instead of \n"